src/waveform/waveform.h

```diff
@@ -143,8 +151,8 @@ class Waveform {
 
     // If stored in the database, the ID of the waveform.
     int m_id;
-    // AnalysisDAO needs to be able to set the waveform as clean.
-    mutable bool m_bDirty;
+    // mutable since AnalysisDAO needs to be able to set the waveform as saved.
+    mutable SaveState m_saveState;
```

Comment:
Both m_completion and m_saveState should be combined in a single QAtomicInt for tracking the state of a waveform (and not only the "save" state). This integer might be encoded as a bitmask if the state is a combination of orthogonal aspects.